rust/lance/src/dataset.rs

```diff
@@ -812,51 +815,84 @@ impl Dataset {
         Ok(counts.iter().sum())
     }
 
-    pub async fn take(&self, row_indices: &[usize], projection: &Schema) -> Result<RecordBatch> {
-        let mut sorted_indices: Vec<u32> =
-            Vec::from_iter(row_indices.iter().map(|indice| *indice as u32));
-        sorted_indices.sort();
+    pub async fn take(&self, row_indices: &[u64], projection: &Schema) -> Result<RecordBatch> {
+        let mut sorted_indices: Vec<usize> = (0..row_indices.len()).collect();
```

Comment:
Minor but do you want to return early if `row_indices` is empty?  Mainly this would allow `take` with zero indices to work even on an empty dataset.  Though I'm sure that's a corner case.

Reply:
Not a bad idea 👍